File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        int top = -1, unbalanced = 0;
        for (char bracket : s.toCharArray()) {
            if (bracket == '[') {
                top += 1;
            } else {
                if (top == -1) {
                    unbalanced += 1;
                } else {
                    top -= 1;
                }
            }
        }
        int swaps = (unbalanced + 1) / 2;
        return swaps;
    }
}

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.py
class Solution:
    def minSwaps(self, s: str) -> int:
        stack = []
        unbalanced = 0
        for bracket in s:
            if bracket == "[":
                stack.append(bracket)
            else:
                if not stack:
                    unbalanced += 1
                else:
                    stack.pop()

        return (unbalanced + 1) // 2
